File: binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size();

        while (left < right){
            int middle = (right + left) / 2;

            if (target < nums[middle]){
                right = middle;
            }else if (target > nums[middle]){
                left = middle + 1;
            }else{return middle;}
        }

        return -1;
        
    }
};

File: binary-search/binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """

        left = 0
        right = len(nums) - 1

        while left <= right:
            mid = (left + right) // 2

            if target < nums[mid]:
                right = mid - 1
            elif target > nums[mid]:
                left = mid + 1;
            else:
                return mid;
        
        return -1

        